include/openenclave/internal/plugin.h

```diff
@@ -0,0 +1,465 @@
+     * @param[in] opt_params The optional plugin-specific input parameters.
+     * @param[in] opt_params_size The size of opt_params in bytes.
+     * @param[out] evidence_buffer An output pointer that will be assigned the
+     * address of the evidence buffer.
+     * @param[out] evidence_buffer_size A pointer that points to the size of the
+     * evidence buffer in bytes.
+     * @param[out] endorsements_buffer An output pointer that will be assigned
+     * the address of the endorsements buffer.
+     * @param[out] endorsements_buffer_size A pointer that points to the size of
+     * the endorsements buffer in bytes.
+     * @retval OE_OK on success.
+     * @retval An appropriate error code on failure.
+     */
+    oe_result_t (*get_evidence)(
+        oe_attester_t* context,
+        const oe_claim_t* custom_claims,
+        size_t custom_claims_length,
+        const void* opt_params,
+        size_t opt_params_size,
+        uint8_t** evidence_buffer,
+        size_t* evidence_buffer_size,
+        uint8_t** endorsements_buffer,
+        size_t* endorsements_buffer_size);
+
+    /**
+     * Creates a legacy report to be used in local or remote attestation.
+     * The report shall contain the data given by the **report_data** parameter.
+     * This entry point is for the OE SDK framework to implement legacy API
+     * oe_get_report_v2().
+     *
+     * @experimental
+     *
+     * @param[in] context A pointer to the attester plugin struct.
+     * @param[in] flags Specifying default value (0) generates a report for
+     * local attestation. Specifying OE_REPORT_FLAGS_REMOTE_ATTESTATION
+     * generates a report for remote attestation.
+     * @param[in] report_data The report data that will be included in the
+     * report.
+     * @param[in] report_data_size The size of the **report_data** in bytes.
+     * @param[in] opt_params Optional additional parameters needed for the
+     * current enclave type. For SGX, this can be sgx_target_info_t for local
+     * attestation.
+     * @param[in] opt_params_size The size of the **opt_params** buffer.
+     * @param[out] report_buffer This points to the resulting report upon
+     * success.
+     * @param[out] report_buffer_size This is set to the
+     * size of the report buffer on success.
+     *
+     * @retval OE_OK The report was successfully created.
+     * @retval OE_INVALID_PARAMETER At least one parameter is invalid.
+     * @retval OE_OUT_OF_MEMORY Failed to allocate memory.
+     * @retval An appropriate error code on failure.
+     *
+     */
+    oe_result_t (*get_report)(
+        oe_attester_t* context,
+        uint32_t flags,
+        const uint8_t* report_data,
+        size_t report_data_size,
+        const void* opt_params,
+        size_t opt_params_size,
+        uint8_t** report_buffer,
+        size_t* report_buffer_size);
+
+    /**
+     * Frees the generated attestation evidence and endorsements.
```

Comment:
The "and endorsements" part is confusing given the separate API at line 185.
```suggestion
     * Frees the generated attestation evidence.
```

Reply:
Thanks!